clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocol.java

```diff
@@ -78,6 +78,8 @@
     public static final Schema ASSIGNMENT_V0 = new Schema(
             new Field(TOPIC_PARTITIONS_KEY_NAME, new ArrayOf(TOPIC_ASSIGNMENT_V0)),
             new Field(USER_DATA_KEY_NAME, Type.NULLABLE_BYTES));
+    public static final Schema TOPIC_PARTITION_ASSIGNMENT_V0 = new Schema(
+            new Field(TOPIC_PARTITIONS_KEY_NAME, new ArrayOf(TOPIC_ASSIGNMENT_V0)));
```

Comment:
Hmm.. I was expecting that the serialization would be custom to `StickyAssignor`. We have the UserData field specifically to let assignors embed their own metadata. So I would expect that we move the assignment schema into `StickyAssignor` and add logic to deserialize the UserData field when we receive the subscriptions. Does that make sense?